torchx/schedulers/slurm_scheduler.py

```diff
@@ -482,10 +488,82 @@ def _cancel_existing(self, app_id: str) -> None:
         subprocess.run(["scancel", app_id], check=True)
 
     def describe(self, app_id: str) -> Optional[DescribeAppResponse]:
-        try:
-            return self._describe_sacct(app_id)
-        except subprocess.CalledProcessError:
-            return self._describe_squeue(app_id)
+        # fallback to using different slurm commands for describing the job
+        for describe in [
+            self._describe_scontrol,  # NOTE: only scontrol fills hostnames
+            self._describe_sacct,
+            self._describe_squeue,
+        ]:
+            try:
+                return describe(app_id)
+            except CalledProcessError:
+                continue
+
+    def _describe_scontrol(self, app_id: str) -> Optional[DescribeAppResponse]:
```

Comment:
`scontrol show --json job 221` and `squeue --json -j 221` appear to be identical outputs. Can we just merge this into the existing `_describe_squeue` method below?

Reply:
done